src/iexit.cpp

```diff
@@ -26,12 +35,39 @@ main( int argc, char **argv ) {
     rodsEnv myEnv;
     getRodsEnv( &myEnv );
     char* envFile = getRodsEnvFileName();
-    status = unlink( envFile );
-    if ( myRodsArgs.verbose == True ) {
-        printf( "Deleting (if it exists) session envFile: %s\n", envFile );
-        printf( "unlink status [%d]\n", status );
-        // prompt before removing
-        obfRmPw( 0 );
+    const bool isServiceAccount = irods::is_service_account();
+
+    status = unlink(envFile);
```

Comment:
Do we want to wrap the `unlink` itself with `!isServiceAccount`?

That would prevent the environment file from being deleted too... 

I'm feeling a markdown table itching to get out...

Reply:
I don't quite understand the logic going on behind the scenes, but when `iexit` is run as the service account, it doesn't try to remove `irods_environment.json`, but `irods_environment.json.[SOME NUMBER]`

Reply:
oooh, right.   because there could be multiple connections/sessions for the same user.

carry on.

Reply:
Should the message on 42 come before the `unlink` on 40? I realize that would introduce 2 if blocks, just seems upside-down to me.